optaplanner-examples/src/main/java/org/optaplanner/examples/nqueens/app/NQueensApp.java

```diff
@@ -62,7 +62,6 @@ public NQueensApp() {
                 NQueensPanel.LOGO_PATH);
     }
 
-    @Override
```

Comment:
BUG (important)

This doesn't work any more, because the super method was removed and it no longer passed through this.
So changing this method to switch to createSolverFactoryByApi() won't actually do anything.

Do a find-in-path where createSolverFactoryByApi() and createSolverFactoryByXml() is mentioned in the docs, comments, etc.

Proposals:
A) Revert these changes.
B) Introduce a createSolverConfig() method that can be overwritten?
C) Stop supporting createSolverFactoryByApi() in NQueensApp in bring it inline with the other examples. Adjust all docs accordingly.

Either way, look at NQueensApp.createSolverFactoryByApi() and compare that with TimeTableApp in our optaplanner-quickstarts hello-world:

```
        SolverFactory<TimeTable> solverFactory = SolverFactory.create(new SolverConfig()
                .withSolutionClass(TimeTable.class)
                .withEntityClasses(Lesson.class)
                .withConstraintProviderClass(TimeTableConstraintProvider.class)
                // The solver runs only for 5 seconds on this small dataset.
                // It's recommended to run for at least 5 minutes ("5m") otherwise.
                .withTerminationSpentLimit(Duration.ofSeconds(5)));
```

That is much better, so maybe C) is the path to go?

Reply:
This change was not necessary and honestly I didn't even know I made it.
I reverted to the original state, keeping the status quo.

Reply:
Agreed, the optaplanner-quickstart hello-world example covers this much better.